org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/viewer/MavenTypeEditingSupport.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Patrick Ziegler and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.pde.ui.target.editor.internal.viewer;
+
+import java.util.List;
+
+import org.eclipse.jface.viewers.CellEditor;
+import org.eclipse.jface.viewers.ColumnViewer;
+import org.eclipse.jface.viewers.ComboBoxCellEditor;
+import org.eclipse.jface.viewers.EditingSupport;
+import org.eclipse.m2e.pde.target.MavenTargetDependency;
+import org.eclipse.m2e.pde.ui.target.editor.internal.Constants;
+import org.eclipse.m2e.pde.ui.target.editor.internal.MavenAccessor;
+import org.eclipse.swt.widgets.Composite;
+
+/**
+ * <p>
+ * This class provides the ability to edit the type of a Maven dependency via a
+ * combo box. Supported types are:
+ * <ul>
+ * <li>jar</li>
+ * <li>bundle</li>
+ * <li>pom</li>
+ * </ul>
+ * </p>
+ * 
+ * @see Constants#TYPES
+ */
+public class MavenTypeEditingSupport extends EditingSupport {
+	private final MavenAccessor accessor;
+
+	public MavenTypeEditingSupport(ColumnViewer viewer) {
+		super(viewer);
+		this.accessor = MavenAccessor.forType();
+	}
+
+	@Override
+	protected CellEditor getCellEditor(Object element) {
+		return new ComboBoxCellEditor((Composite) getViewer().getControl(), Constants.TYPES);
+	}
+
+	@Override
+	protected boolean canEdit(Object element) {
+		return element instanceof MavenTargetDependency;
+	}
+
+	@Override
+	protected Object getValue(Object element) {
+		String value = accessor.apply(element);
```

Comment:
Do we really need to operate on the index? What about the user is entering a custom type (what is allowed)